test/lib/DuplexCollection.js

```diff
@@ -0,0 +1,138 @@
+	t.deepEqual(duplexCollection._combo._readers, [{}, {}], "reader and writer assigned to readers");
+});
+
+test("DuplexCollection: _byGlob", (t) => {
+	t.plan(2);
+
+	const abstractReader = {
+		_byGlob: sinon.stub().returns(Promise.resolve([]))
+	};
+
+	const duplexCollection = new DuplexCollection({
+		name: "myCollection",
+		reader: abstractReader,
+		writer: abstractReader
+	});
+
+	const trace = {
+		collection: sinon.spy()
+	};
+
+	const comboSpy = sinon.spy(duplexCollection._combo, "_byGlob");
+
+	return duplexCollection._byGlob("anyPattern", {someOption: true}, trace)
+		.then(function(resources) {
+			t.true(Array.isArray(resources), "Found resources are returned as an array");
+			t.true(comboSpy.calledWithExactly("anyPattern", {someOption: true}, trace),
+				"Delegated globbing task correctly to readers");
+		});
+});
+
+test("DuplexCollection: _byPath with reader finding a resource", (t) => {
+	t.plan(4);
+
+	const resource = new Resource({
+		path: "path",
+		buffer: new Buffer("content")
+	});
+
+	const pushCollectionSpy = sinon.spy(resource, "pushCollection");
+
+	const abstractReader = {
+		_byPath: sinon.stub().returns(Promise.resolve(resource))
+	};
+
+	const trace = {
+		collection: sinon.spy()
+	};
+
+	const duplexCollection = new DuplexCollection({
+		name: "myCollection",
+		reader: abstractReader,
+		writer: abstractReader
+	});
+
+	const comboSpy = sinon.spy(duplexCollection._combo, "_byPath");
+
+	return duplexCollection._byPath("anyVirtualPath", {someOption: true}, trace)
+		.then(function(resource) {
+			t.true(comboSpy.calledWithExactly("anyVirtualPath", {someOption: true}, trace),
+				"Delegated globbing task correctly to readers");
+			t.true(pushCollectionSpy.called, "pushCollection called on resource");
+			resource.getString().then(function(content) {
+				t.deepEqual(content, "content", "Resource has expected content");
+			});
+			t.deepEqual(resource.getPath(), "path", "Resource has expected path");
+		});
+});
+
+test("DuplexCollection: _byPath with two readers both finding no resource", (t) => {
+	t.plan(3);
+
+	const abstractReaderOne = {
+		_byPath: sinon.stub().returns(Promise.resolve())
+	};
+
+	const abstractReaderTwo = {
+		_byPath: sinon.stub().returns(Promise.resolve())
+	};
+
+	const trace = {
+		collection: sinon.stub()
+	};
+
+	const duplexCollection = new DuplexCollection({
+		name: "myCollection",
+		reader: abstractReaderOne,
+		writer: abstractReaderTwo
+	});
+
+	return duplexCollection._byPath("anyVirtualPath", {someOption: true}, trace)
+		.then(function(resource) {
+			t.true(abstractReaderOne._byPath.calledWithExactly("anyVirtualPath", {someOption: true}, trace),
+				"Delegated globbing task correctly to reader one");
+			t.true(abstractReaderTwo._byPath.calledWithExactly("anyVirtualPath", {someOption: true}, trace),
+				"Delegated globbing task correctly to reader two");
+			t.falsy(resource, null, "No resource found");
```

Comment:
falsy only accepts two arguments. You can remove the `null`

Reply:
oh. done.